Sorbet/main.rb

```diff
@@ -0,0 +1,326 @@
+# typed: strict
+extend T::Sig
+
+# Example positive
+sig { params(x: T.untyped).returns(T.untyped) }
+def positive_success_f(x)
+  if x.is_a?(String)
+    x.length
+  else
+    x
+  end
+end
+
+sig { params(x: T.untyped).returns(T.untyped) }
+def positive_failure_f(x)
+  if x.is_a?(String)
+    x.is_nan # Expected error: No method 'is_nan' on String
+  else
+    x
+  end
+end
+
+# Example negative
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def negative_success_f(x)
+  if x.is_a?(String)
+    x.length
+  else
+    x + 1
+  end
+end
+
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(Integer) }
+def negative_failure_f(x)
+  if x.is_a?(String)
+    x.length
+  else
+    x + 1 # Expected error: + not defined for TrueClass | FalseClass
+  end
+end
+
+# Example alias
+sig { params(x: T.untyped).returns(T.untyped) }
+def alias_success_f(x)
+  y = x.is_a?(String)
+  if y
+    x.length
+  else
+    x
+  end
+end
+
+sig { params(x: T.untyped).returns(T.untyped) }
+def alias_failure_f(x)
+  y = x.is_a?(String)
+  if y
+    x.is_nan # Expected error: No method 'is_nan' on String
+  else
+    x
+  end
+end
+
+sig { params(x: T.untyped).returns(T.untyped) }
+def alias_failure_g(x)
+  y = x.is_a?(String)
+  y = true
+  if y
+    x.length # Expected error: x not guaranteed to be String
+  else
+    x
+  end
+end
+
+# Example connectives
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def connectives_success_f(x)
+  if !x.is_a?(Integer)
+    x.length
+  else
+    0
+  end
+end
+
+sig { params(x: T.untyped).returns(Integer) }
+def connectives_success_g(x)
+  if x.is_a?(String) || x.is_a?(Integer)
+    connectives_success_f(x)
+  else
+    0
+  end
+end
+
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(Integer) }
+def connectives_success_h(x)
+  if x.is_a?(String)
+    x.length
+  else
+    0
+  end
+end
+
+sig { params(x: T.any(String, Integer)).returns(Integer) }
+def connectives_failure_f(x)
+  if !x.is_a?(Integer)
+    x.is_nan # Expected error: is_nan not defined for String
+  else
+    0
+  end
+end
+
+sig { params(x: T.untyped).returns(Integer) }
+def connectives_failure_g(x)
+  if x.is_a?(String) || x.is_a?(Integer)
+    x.length # Expected error: length not defined for String | Integer
+  else
+    0
+  end
+end
+
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(Integer) }
+def connectives_failure_h(x)
+  if x.is_a?(String)
+    x.length
+  else
+    x.length # Expected error: length not defined for Integer | TrueClass | FalseClass
+  end
+end
+
+# Example nesting_body
+sig { params(x: T.any(String, Integer, TrueClass, FalseClass)).returns(Integer) }
+def nesting_body_success_f(x)
+  if x.is_a?(Integer)
```

Comment:
Wrong condition, should use nested ifs.
https://github.com/utahplt/ifT-benchmark?tab=readme-ov-file#success-expected-3